### PR TITLE
Add learning rate properties to gluon trainer and optimizer

### PR DESCRIPTION
**Pull request by astonzhang**

### CODE REVIEW

File: python/mxnet/optimizer.py [hunk 1]
```diff
@@ -194,6 +209,15 @@ def update(self, index, weight, grad, state):
         """
         raise NotImplementedError()
 
+    @learning_rate.setter
+    def learning_rate(self, lr):
```
Comment (piiswrong) [inline python/mxnet/optimizer.py]: Let's use set_learning_rate explicitly rather than a property setter.

Comment (piiswrong) [reply python/mxnet/optimizer.py]: and add documentation to the function

Comment (astonzhang) [reply python/mxnet/optimizer.py]: resolved

File: python/mxnet/optimizer.py [hunk 2]
```diff
@@ -154,6 +161,12 @@ def create_optimizer(name, **kwargs):
         else:
             raise ValueError('Cannot find optimizer %s' % name)
 
+    @property
+    def learning_rate(self):
+        if self.lr_scheduler is not None:
+            return self.lr_scheduler.base_lr
```
Comment (piiswrong) [inline python/mxnet/optimizer.py]: self.lr_scheduler(self.num_update)

Comment (astonzhang) [reply python/mxnet/optimizer.py]: resolved
